proposals/0072-feature-gate-threshold-automation.md

```diff
@@ -0,0 +1,248 @@
+`SignalSupportForStagedFeature` instructions.
+
+```
+Instruction: SignalSupportForStagedFeature
+- Data:
+    - 1 byte: Instruction discriminator
+- Accounts:
+    - [s]: Authorized voter
+    - [ ]: Validator vote account
+    - [w]: Signal account
+    - [w]: Feature account
+```
+
+The authorized voter signer must match the authorized voter stored in the vote
+account's state.
+
+The program processor will use the `SolGetEpochStake` syscall to retrive the
+corresponding epoch stake for the provided vote account, and add it to the
+feature's stake support.
+
+The signal account is a PDA derived from the epoch, feature ID and vote address
+to prevent double-signaling.
+
+```
+epoch_le_bytes + feature_id + vote_address
+```
+
+It contains just 8 bytes for the little-endian `u64` stake amount. These signal
+accounts can serve as historical records for which nodes signaled support for
+which features. They also serve as a source of truth for withdrawing support via
+the `WithdrawSupportForStagedFeature` instruction, which will deduct the stored
+stake value in the signal account from the feature's stake support and clear the
+signal account.
+
+```
+Instruction: WithdrawSupportForStagedFeature
+- Data:
+    - 1 byte: Instruction discriminator
+- Accounts:
+    - [s]: Authorized voter
+    - [ ]: Validator vote account
+    - [w]: Signal account
+    - [w]: Feature account
+```
+
+Although there is no requirement for features to be staged in the previous
+epoch, staging might occur late in an epoch, after most support signals have
+already been cast. This would result in not enough stake support until the next
+epoch arrives and new signals are cast.
+
+### Activation
+
+At the end of the epoch, the runtime loads all staged feature accounts and
+calculates their stake support as a percentage of the cluster-wide stake. Only
+features whose account state has status `Staged` at the epoch rollover will be
+evaluated. 
+
+Every feature whose stake support meets the required threshold will be
+activated. This threshold will be hard-coded in the runtime to 95% initially,
+but future iterations on the process could make this threshold configurable.
+
+Features without the required stake support will remain `Staged` until they
+acquire enough stake support or are manually unstaged.
+
+## Alternatives Considered
+
+## Impact
+
+This new process for activating features directly impacts core contributors and
+validators.
+
+Core contributors will no longer bear the responsibility of ensuring the proper
+stake supports their feature activation. However, this proposal does not include
+a mechanism for overriding or customizing the stake requirement. This capability
+should be proposed separately.
+
+Validators will be responsible for signaling their vote using a transaction
+which they've previously not included in their process. They also will have a
+more significant impact on feature activations if they neglect to upgrade their
+software version.
```

Comment:
nit: I don't think they have any more or less of an impact on feature activations after this proposal